app/controllers/reports_controller.rb

```diff
@@ -21,6 +21,9 @@ def index
   def show
     @products = @report.user.products.not_wip.order(published_at: :desc)
     @recent_reports = Report.list.where(user_id: @report.user.id).limit(10)
+    Footprint.create_on_resource(@report, current_user) unless @report.user == current_user
```

Comment:
```suggestion
    Footprint.find_or_create_by(footprintable: resource, user: current_user) unless @report.user == current_user
```

ラップしない方が読む人にとってわかりやすいように思いました。

Reply:
おっしゃる通りです
修正いただいたコードの方が直感的に読めます